docs/source/reference/storage.rst

```diff
@@ -204,100 +134,142 @@ and storage mounting:
     aws/gsutil cli) will see the latest data.
 
 .. note::
-    :code:`MOUNT` mode does not support the full POSIX interface and some file
-    operations may fail. Most notably, random writes and append operations are
-    not supported.
+    SkyPilot does not guarantee preservation of file permissions when attaching
+    object stores. You may need to set file permissions during task execution.
 
 .. note::
-    Storage only supports uploading directories (i.e., :code:`source` cannot be a file).
-    To upload a single file to a bucket, please put in a directory and specify the directory as the source.
-    To directly copy a file to a VM, please use regular :ref:`file mounts <file-mounts-example>`.
+    Symbolic links are handled differently in :code:`file_mounts` depending on whether object stores are used.
+    For object store mounts, symbolic links are not copied to remote.
+    For local ``file_mounts`` that are directly rsynced to the VM,
+    the symbolic links are directly copied, not their target data.
+    The targets must be separately mounted or else the symlinks may break.
 
-Creating a shared file system
------------------------------
+Common patterns
```

Comment:
I guess it would be nice to mention the ML specific use case in each of the following pattern in this section, e.g., `Storing and sharing outputs of tasks (model checkpoints)`, `Avoid re-uploading data on every run (uploading and sharing artifacts for multiple runs)`, `Copying data from storage before job starts (Dataset)` etc.

Reply:
good idea - I've added a **💡 Example use-case** under each pattern. does that help?